gestalt/vault.py

```diff
@@ -243,3 +246,41 @@ def worker(self, token_queue: Queue) -> None:  # type: ignore
     @property
     def scheme(self) -> str:
         return self._scheme
+
+    def _validate_token_expiration(self) -> None:
+        token_details = self.vault_client.auth.token.lookup_self()
+        if token_details['data'] is not None:
+
+            expire_time = None
+            if 'expire_time' not in token_details['data']:
+                print(
+                    "Key 'expire_time' does not exist in token_details['data']"
+                )
+                return None
+
+            expire_time = token_details['data']['expire_time']
+            # Validate expire_time is present
+            if expire_time is None:
+                print("Cannot parse expire_time, value is None")
+                return None
+
+            # Use isoparse to correctly parse the datetime string
+            expire_time = isoparse(expire_time)
```

Comment:
can there be a parsing error here?

Reply:
Yes, the isoparse lib handle that and raise an exception for it.